Readme.asciidoc

```diff
@@ -1,68 +1,116 @@
+=== Creating a client application
 
-Once it's done, just use a Hot Rod client (or REST) to connect to it. Below is an example:
+Once you instantiate one of the services, you might want to connect to it using a Hot Rod (or REST) client.
 
-```
-Configuration cachingServiceClientConfiguration = new ConfigurationBuilder()
-   .addServer()
-      .host("infinispan-app-hotrod") // This service is created by the template
-      .port(11222) // Standard Hot Rod port
-   .build();
+The easiest way to do it is to use https://github.com/jboss-developer/jboss-jdg-quickstarts/tree/jdg-7.2.x/caching-service[JDG Caching Service Quickstart]. The most important thing to remember is that
+ all services use authentication. The same credentials that were used while creating a service need to be used
+ in the client application. Here's an example:
 
-RemoteCacheManager cachingService = new RemoteCacheManager(cachingServiceClientConfiguration);
-RemoteCache<String, String> defaultCache = cachingService.getCache();
 
-defaultCache.put("test", "test"); // Start playing with the Cache
 ```
+Configuration c = new ConfigurationBuilder()
+      .addServer()
+          .host("caching-service-app")
+          .security().authentication().enable()
+            .username("test")
+            .password("test")
+            .realm("ApplicationRealm")
+            .serverName("caching-service")
+            .saslMechanism("DIGEST-MD5")
+            .saslQop(SaslQop.AUTH)
+      .build();
+
+RemoteCacheManager remoteCacheManager = new RemoteCacheManager(c);
+RemoteCache<String, String> cache = remoteCacheManager.getCache();
+```
+
+The above template might be used only as an example. The quickstart covers more details on connecting to the existing
+ JDG Online Service.
 
-What's next?
-------------
+== What's next?
 
 If you're interested in our plans, visit our https://issues.jboss.org/secure/RapidBoard.jspa?rapidView=4167[JIRA board].
 
-Useful links (mostly for development)
--------------------------------------
+== Development guide
+
+Prior to exploring this repository, it is highly advised to visit our https://docs.google.com/a/redhat.com/document/d/1gSGmytIW3u4IR1tXGmEekEa1AzhKfNvAPOAP0Fh32kQ/edit?usp=sharing[Contributing Guide].
+
+This repository has the following structure:
+
+```
++- Root
+      +-- image.yaml - Image build recipe
+      +-- Makefile - A set of useful scripts for development work
+      +-- templates - OpenShift templates for this service
+      +-- modules - Modules used for the image
+      +-- functional-tests - Full end-to-end tests with Arquillian Cube
+```
+
+The `image.yaml` uses Python framework called http://concreate.readthedocs.io/en/develop/[Concreate] to build an image. The easiest way to start the build is to invoke `concreate image.yaml target` and invoke a standard Docker build from `target` directory.
```

Comment:
"The `image.yaml` uses *a* Python framework called..."

"...and invoke a standard Docker build from *the* `target` directory.

Reply:
Done